modules/generator/registry/histogram.go

```diff
@@ -147,7 +149,11 @@ func (h *histogram) updateSeries(s *histogramSeries, value float64, traceID stri
 	s.lastUpdated.Store(time.Now().UnixMilli())
 }
 
-func (h *histogram) collectMetrics(appender storage.Appender, timeMs int64, externalLabels map[string]string) (activeSeries int, err error) {
+func (h *histogram) Name() string {
+	return h.name
+}
+
+func (h *histogram) CollectMetrics(appender storage.Appender, timeMs int64, externalLabels map[string]string) (activeSeries int, err error) {
```

Comment:
any particular reason these were exported from the package?

Reply:
Err, for consistency and to avoid naming conflicts 😅  The interface and the structs these methods are linked to aren't exported, so I didn't think it matters much.

But basically:
- I introduced the method `name()` to the `metric` interface
- this conflicts with the field `name`, so I decided to rename the function to `Name`
- for consistency, I've also written the other functions with a capital letter

https://github.com/kvrhdn/tempo/blob/23ebaf43bd3dfce386a23f965ca42ab9d09ec457/modules/generator/registry/registry.go#L74-L78

I can see if there is a cleaner solution...

Reply:
Alright, reverted back the name change.